### ISSUE TITLE
[Doc] Add section to "ci-fix.md" for "Different behavior in CI and locally"

### ISSUE THREAD

**Issue by mikeharder** [Azure/azure-rest-api-specs#28796]
1. Merge latest from parent (e.g. `main` or `RPSaaSMaster`) to PR branch
2. Run locally
   1.  `git clean -xdf` (may revert any local changes)
   2. `git pull`
   3. `npm ci`
3. `npx tsv ...`

**Comment by TheDonLuke** [comment]
I'm seeing errors on GitHub that I cannot reproduce locally after running the above commands. https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3856615&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=0f4aa080-c997-58ca-481c-4930865ae0f8&l=59

**Comment by mikeharder** [comment]
> I'm seeing errors on GitHub that I cannot reproduce locally after running the above commands. https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3856615&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=0f4aa080-c997-58ca-481c-4930865ae0f8&l=59 for PR [Azure/azure-rest-api-specs-pr#18520](https://github.com/Azure/azure-rest-api-specs-pr/pull/18520)

@TheDonLuke: Works for me:

```
$ gh pr checkout 18520
$ git clean -xdf
$ npm ci
$ npx tsv specification/fist/IotFirmwareDefense.Management

$ git status
On branch RPSaaSDev
Your branch is ahead of 'origin/RPSaaSDev' by 5 commits.
  (use "git push" to publish your local commits)

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        specification/fist/resource-manager/Microsoft.IotFirmwareDefense/
        
$ ls specification/fist/resource-manager/
Microsoft.IotFirmwareDefense  Private.IoTFirmwareDefense  readme.cli.md     readme.go.md  readme.python.md
Microsoft.IoTFirmwareDefense  readme.az.md                readme.csharp.md  readme.md     readme.typescript.md
```

Looks like you have a bug.  There are two folders that only differ in case.

**Comment by TheDonLuke** [comment]
Ah, I see.  the tsv tool doesn't report differences on windows (native or wsl) or on macos.  We were able to repro successfully if we used an actual linux machine

**Comment by mikeharder** [comment]
> Ah, I see. the tsv tool doesn't report differences on windows (native or wsl) or on macos. We were able to repro successfully if we used an actual linux machine

I don't think this is related to `tsv`, or there is anything `tsv` could do better.  The different OS filesystems have different casing rules, which can cause different impacts if you mix cases in your code/config.

**Comment by mikeharder** [comment]
Moved to https://github.com/Azure/azure-rest-api-specs/wiki/TypeSpec-Validation
